packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -86,18 +88,19 @@ export const DataLinkInput: React.FC<DataLinkInputProps> = memo(({ value, onChan
     }
   }, []);
 
+  useEffect(() => {
```

Comment:
Works, but a bit weird that we have to track blur like this.

Reply:
Well the main point is we mix imperative editor control with controlling the value of the Editor. We should probably stick to just one. Like here we probably do not need to have Value in the local state at all and just use editor commands instead (this seems like better documented by slate).

Other way would be to go only by state changes but then we would need a way to easily trigger changes with some help utils (like `setFocused(value)` and `isFocused(value)`) but slate does not seem to have such.

So in essence this is "React" way to do this, handlers and commands due to their implementation is more imperative slate way to do it. For example the onBlur handler is changing the state of the Editor, "React" way would be to do nothing and let the component change the local Value and pass it to the Editor with `focus=false`.